### DISCUSSION TITLE
Why MuseScore 3 continues to be my music notation, for now

### DISCUSSION THREAD

**Discussion by cleyburn** [musescore/MuseScore#15751] [General]
After using MS4 for the gamut of composing, transcribing, playing and exporting, here is my current assessment of its usability for my needs. So much has been crowed about MS4's limited music production prowess that I don't need to repeat it in this critique, except to say that MS4 voices have many holes I'm sure that you are aware of and I am confident that you will concentrate on filling them in. However, I do not believe that music/sound production should be the sole focus, or even the primary focus of your immediate work. After several refinements through the years and with the addition of a few very versatile and important plug-ins, MS3 was a very useful music transcription tool with which I could experiment and compose original music of many parts up to orchestral size or transcribe simple jazz lead sheets with the speed and flexibility to get things done quickly and easily, - and format a very nice, printed and even audible output. MS4 no longer fulfills that need for music transcription software that was the original, primary goal of MuseScore. I have found that the most basic act of single note entry has become a time-consuming and awkward hassle, let alone trying to enter chords in one pass cleanly - which near impossible. (I am not willing to accept the criticism that my cpu's lack of a dozen cores is to blame for the latency. If you hope to create free software for the people of the world, you can’t play the Microsoft “glutware” game. The people of the world just can’t afford to upgrade their equipment that quickly. MS4 just has to be much, much more optimized, smaller and more efficient.) Here are a couple of items I think you should focus on immediately: 
1.)	Improve the ease of note entry. Bring it back to MS3 standards. Why can’t you de-couple the sophisticated sound production while performing note entry? Just use those complex systems when playing back what’s written? But while entering notes I should have absolutely no latency *and* I should be able to play polyphonic midi from my keyboard without note entry even active (it’s called composing), something I can no longer do in MS4. For that reason, MS3 is still my primary composition tool. (I sometimes use MS4 for sound production, but only after very carefully using separately named and isolated files for the purpose.) 
2.)	I find the new MS4 interface slightly improved, but bulky and less versatile for efficient screen use. I should be able to detach and move those tabs anywhere I like. (I use four screens.) I should also be able to size those windows down, too. They’re just too bulky.
3.)	Plug-ins are very important to my music composition. I’m pleased to see mirror intervals, retrograde, and even courtesy accidental tools offered. But I also miss, very much, the plug-ins I used for checking against traditional harmonic rules, including finding parallel fifths, octaves, augmented fourths and the like. Please, please provide a complete replacement tool for these functions.
For now, please do nothing new until you bring back what we’ve lost from MS3. After all, since the file format is not compatible, MS4 is just not usable for me until you do.
I hope you find my observations and desires are valuable. I know you’ll get there, and I promise to be patient. And I can’t thank you enough for the work you’ve done creating and improving this amazing tool for artists worldwide. Keep up the great work.

**Comment by cbjeukendrup** [comment]
- Note input is still a bit laggy -> yes, there is still room for optimisation, seemingly especially relating to playback and especially Muse Sounds
(On the other hand, "optimising" is much easier said than done, and can't be done infinitely. It might turn out that to be physically impossible to get the very best playback on low-end processors, in which case you'll have to switch to MS Basic playback, which can be done anyway in View > Playback Setup > MS Basic > "Activate this profile". MS Basic is the same playback as in MS3 by the way. Also, it might help to go to Preferences > Note input and disable "Play notes during editing". However, you really aren't supposed to need 12 cores for MuseScore, so we are definitely continuing to work on this.)
- Playing on a MIDI keyboard still has some bugs -> yes, known issues, https://github.com/musescore/MuseScore/issues/14673 and https://github.com/musescore/MuseScore/issues/13381
- The minimum window width is inconveniently big -> yes, known issue, https://github.com/musescore/MuseScore/issues/13372, but fixing this is less trivial than it seems, so therefore it was not fixed yet for 4.0.0
- "I should be able to detach and move those tabs anywhere I like." -> What do you mean with this one? Toolbars and palettes/instruments/properties panels can be detached from the main window and moved anywhere. 
- You want to be able to resize panels (to a smaller size) -> also something we want to support again, but is currently disabled because it caused trouble
- You relied a lot upon plugins -> the plugin framework from MS3 is not officially supported anymore, since it is outdated and limited. We are looking into designing a completely new, much more powerful framework, but this takes time. For the time being, the existing framework is still present in MS4, so many plugins from MS3 should still run fine, especially those plugins that work within the current score, rather than opening/saving/exporting files. For plugins like checking for parallel fifths, you should definitely give it a try. Just put them in the folder specified in Preferences > General > Folders, and they will show up in the Home tab, where you can enable them. 

So indeed, we'll get there! The things you've pointed out are valid and we're aware of them.

**Comment by cleyburn** [comment]
Thank you, Casper. You guys are wonderful. I just need to hang tight.
Meanwhile, it's all very exciting!

Sincerely yours,

♫ь♪ Bruce DeLoria ♯♪♫
US Phones & WhatsApp: 503 389-0630
   Tillamook Landline: 503 815-8098

*“Le roi ne lache que quand le peuple arrache” -Victor Hugo, 1831*

*[The King only grants what the people wrest from him.] *



KUVO Jazz <http://kuvo-ice.streamguys.org/kuvo-aac-128> - Radio
<http://kuvo-ice.streamguys.org/kuvo-aac-128>Portland
<https://player.allclassical.org/>All <https://player.allclassical.org/>
Classical <https://player.allclassical.org/> - Luxuria Radio
<https://luxuriamusic.com/luxuriamusic_player/>

**Comment by aige02** [comment]
Since I enter notes from the computer keyboard with the sound turned off, I haven't experienced @cleyburn's problems with latency when entering music.  I have found that bringing up the massive *Style* menu has too much latency.  The first time I tried, I though MU 4.0 had hung!  Now I know to just relax for a while - it will eventually appear.

As much as I do like MU 4.0, I only use MU 4.0 for what I consider Beta testing.  Much of my MU work is with Big Band scores - about 17 to 25 staves per system and usually at least 17 parts.  I have found and reported several critical bugs, some of which caused crashes, particularly when parts have been created.  The most serious problem I am facing is that individual parts cannot be printed or saved as a PDF (see [issue 16537](https://github.com/musescore/MuseScore/issues/16537)).  Because of problems getting the parts formatted, I haven't yet tried saving either the score or all the parts as a PDF.
